tracing-subscriber/src/fmt/time/chrono_crate.rs

```diff
@@ -0,0 +1,161 @@
+use crate::fmt::format::Writer;
+use crate::fmt::time::FormatTime;
+
+/// Formats [local time]s and [UTC time]s with `FormatTime` implementations
+/// that use the [`chrono` crate].
+///
+/// [local time]: [`chrono::offset::Local`]
+/// [UTC time]: [`chrono::offset::Utc`]
+/// [`chrono` crate]: [`chrono`]
+
+/// The RFC 3339 format is used by default but a custom format string
+/// can be used. See [`chrono::format::strftime`]for details on
+/// the supported syntax.
+///
+/// [`chrono::format::strftime`]: https://docs.rs/chrono/0.4.9/chrono/format/strftime/index.html
+#[cfg_attr(docsrs, doc(cfg(feature = "chrono")))]
+#[derive(Debug, Clone, Eq, PartialEq)]
+enum ChronoFmtType {
+    /// Format according to the RFC 3339 convention.
+    Rfc3339,
+    /// Format according to a custom format string.
+    Custom(String),
+}
+
+impl Default for ChronoFmtType {
+    fn default() -> Self {
+        ChronoFmtType::Rfc3339
+    }
+}
+
+/// Formats the current [local time] using a [formatter] from the [`chrono`] crate.
+///
+/// [local time]: chrono::Local::now()
+/// [formatter]: chrono::format
+#[cfg_attr(docsrs, doc(cfg(feature = "chrono")))]
+#[derive(Debug, Clone, Eq, PartialEq, Default)]
+pub struct ChronoLocal {
+    format: ChronoFmtType,
+}
+
+impl ChronoLocal {
+    /// Format the time using the [`RFC 3339`] format
+    /// (a subset of [`ISO 8601`]).
+    ///
+    /// [`RFC 3339`]: https://tools.ietf.org/html/rfc3339
+    /// [`ISO 8601`]: https://en.wikipedia.org/wiki/ISO_8601
+    pub fn rfc_3339() -> Self {
+        Self {
+            format: ChronoFmtType::Rfc3339,
+        }
+    }
+
+    /// Format the time using the given format string.
+    ///
+    /// See [`chrono::format::strftime`] for details on the supported syntax.
+    pub fn new(format_string: String) -> Self {
+        Self {
+            format: ChronoFmtType::Custom(format_string),
+        }
+    }
+}
+
+impl FormatTime for ChronoLocal {
+    fn format_time(&self, w: &mut Writer<'_>) -> alloc::fmt::Result {
+        let t = chrono::Local::now();
+        match &self.format {
+            ChronoFmtType::Rfc3339 => w.write_str(&t.to_rfc3339()),
```

Comment:
`to_rfc3339()` returns a `String`, so this allocates a new `String`, which lasts only as long as the call to `w.write_str()`, and then immediately drops the `String`. we should try to avoid allocations of temporary strings in code that's run for every log event, as this has unacceptable performance overhead.

There should be a way to use the `chrono` API to produce a [`DelayedFormat`](https://docs.rs/chrono/latest/chrono/format/struct.DelayedFormat.html) that will format the timestamp in RFC3339, which we can then use with the `write!` macro to write that timestamp directly to the `Writer` without allocating. 

Based on a cursory reading of the `chrono` API documentation, I *believe* that should look something like this:
```suggestion
            ChronoFmtType::Rfc3339 => {
                use core::iter;
                use chrono::format::{Item, Fixed};

                write!(w, "{}", t.format_with_items(iter::once(Item::Fixed(Fixed::RFC3339))))
            }
```